packages/frontend/src/components/SingleSelect/SingleSelectProvider.tsx

```diff
@@ -98,6 +109,22 @@ export const SingleSelectProvider = ({
 
   const { getItemByValue } = useLookupItems({ rawItems: allItems })
 
+  /**
+   * This create the + Add new option at the top
+   */
+  const allItemsWithAddNewOption = useMemo(() => {
+    if (addNew?.type === 'modal') {
+      const addNewByModalItem = {
+        value: '__ADD_NEW_PLACEHOLDER_VALUE__', // this is not referenced anywhere else
```

Comment:
nit: would be nice to add this as a comment in `types.d.ts` e.g.

```
export interface IFieldDropdown extends IBaseField {
  type: 'dropdown'
  showOptionValue?: boolean
  allowArbitrary?: boolean
  // If this is enabled, `__ADD_NEW_PLACEHOLDER_VALUE__` becomes a reserved value.
  addNewOption?: {
    id: DropdownAddNewId // identifier when add new option is selected
    type: DropdownAddNewType
    label: string
  }
```

Reply:
added in SingleSelect/types.ts